lib/Installation/Partitioner/LibstorageNG/v4_3/ExpertPartitionerController.pm

```diff
@@ -140,11 +140,14 @@ sub add_logical_volume {
     $self->get_expert_partitioner_page()->select_volume_group($args->{volume_group});
     $self->get_expert_partitioner_page()->press_add_logical_volume_button();
     $self->get_add_logical_volume_page()->set_logical_volume_name($lv->{name});
+    $self->get_add_logical_volume_page()->set_logical_volume_type($lv->{volume_type});
```

Comment:
I'd like to add here: `if $lv->{volume_type};` because then we know if some parameter is optional and we don't even enter the method or compare anything with 'undef'.

Reply:
That was my first edition, but I changed it when I noticed that inside set_logical_volume_type, there have to be two conditions either way, so it seems a bit unnecessary to have a condition outside. WDYT?

Reply:
advantages: (1) save time (very very little) (2) does not run a function that might be have some bug in the future and there is not need to enter on that function (3) from the caller you can see what is optional and what is not.
The condition inside the function is a switch, it is not related with the condition about if something need to be executed.
disadvantages: (1) perhaps that is a bit uglier to have conditions on your controller, but I don't think it is less readable because reading the controller you know what the test does, so you know just reading that the action that are performed. Still this is a bit of OOP and how will scale, not forcing you to change it if you are not convinced. We might want to discuss this styles offline anytime.

Reply:
ok, makes sense. I have changed it and I will rerun the VRs just in case.

Reply:
I changed my mind again with this :) sorry, but no worries you can leave it like that. The thing is that at the end you only knows what the automation do looking at the test data and the test as much, those lines on the controllers will setup everything, with no conditions or ugly 'if's like those I have introduced and there are also in other parts. A better option is a mix of both things we were talking about and return if undef at the beginning of the function `return unless defined $type` , so we don't go through the switch.
A different case where those 'if's could make sense is for example when you have only one disk and you don't need to go to the tree to select the disk, so you just want to select on the table directly avoid that first action, but in those cases we might still want to create a function that does exactly the same trick checking the undef.